test/unit/pcv/ERC20TokemakPCVDeposit.test.ts

```diff
@@ -0,0 +1,188 @@
+import { expectRevert, balance, getAddresses, getCore } from '../../helpers';
+import { expect } from 'chai';
+import hre, { ethers } from 'hardhat';
+import { Signer } from 'ethers';
+
+const toBN = ethers.BigNumber.from;
+
+describe('ERC20TokemakPCVDeposit', function () {
+  let userAddress: string;
+  let pcvControllerAddress: string;
+  let governorAddress: string;
+
+  const impersonatedSigners: { [key: string]: Signer } = {};
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [
+      addresses.userAddress,
+      addresses.pcvControllerAddress,
+      addresses.governorAddress,
+      addresses.beneficiaryAddress1
+    ];
+
+    for (const address of impersonatedAddresses) {
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [address]
+      });
+
+      impersonatedSigners[address] = await ethers.getSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    ({ userAddress, pcvControllerAddress, governorAddress } = await getAddresses());
+
+    this.core = await getCore();
+    this.token = await (await ethers.getContractFactory('MockERC20')).deploy();
+    this.toke = await (await ethers.getContractFactory('MockERC20')).deploy();
+    this.pool = await (await ethers.getContractFactory('MockTokemakERC20Pool')).deploy(this.token.address);
+    this.rewards = await (await ethers.getContractFactory('MockTokemakRewards')).deploy(this.toke.address);
+    this.deposit = await (
+      await ethers.getContractFactory('ERC20TokemakPCVDeposit')
+    ).deploy(this.core.address, this.pool.address, this.rewards.address);
+
+    this.depositAmount = toBN('10000000000000000000'); // 10 token
+  });
+
+  describe('Deposit', function () {
+    it('reverts if paused', async function () {
+      await this.deposit.connect(impersonatedSigners[governorAddress]).pause({});
```

Comment:
Here and in many places there are empty braces as a param, these can be removed I think